lib/checks/navigation/heading-order-after.js

```diff
@@ -1,15 +1,130 @@
+		ancestry = ancestry.concat(nodePath);
+	}
+
+	return ancestry.join(joinStr);
+}
+
 function headingOrderAfter(results) {
 	if (results.length < 2) {
 		return results;
 	}
 
-	var prevLevel = results[0].data;
+	/**
+	 * In order to correctly return heading order results (even for
+	 * headings that may be out of the current context) we need to
+	 * construct an in-order list of all headings on the page,
+	 * including headings from iframes.
+	 *
+	 * To do this we will find all nested headingOrders (i.e. those
+	 * from iframes) and then determine where those results fit into
+	 * the top-level heading order. once we've put all the heading
+	 * orders into their proper place, we can then determine which
+	 * headings are not in the correct order.
+	 **/
+
+	// start by replacing all array ancestry paths with a flat string
+	// path
+	let headingOrder = results[0].data.headingOrder.map(heading => {
+		return {
+			...heading,
+			ancestry: getFramePath(results[0].node.ancestry, heading.ancestry)
+		};
+	});
+
+	// find all nested headindOrders
+	const nestedResults = results.filter(result => {
+		return result.data && result.data.headingOrder && result.node._fromFrame;
+	});
+
+	// update the path of nodes to include the iframe path
+	nestedResults.forEach(result => {
+		result.data.headingOrder = result.data.headingOrder.map(heading => {
+			return {
+				...heading,
+				ancestry: getFramePath(result.node.ancestry, heading.ancestry)
+			};
+		});
+	});
+
+	/**
+	 * Determine where the iframe results fit into the top-level
+	 * heading order
+	 */
+	function getFrameIndex(result) {
+		const path = getFramePath(result.node.ancestry);
+		const heading = headingOrder.find(heading => {
+			return heading.ancestry === path;
+		});
+		return headingOrder.indexOf(heading);
+	}
+
+	/**
+	 * Replace an iframe placeholder with its results
+	 */
+	function replaceFrameWithResults(index, result) {
+		headingOrder.splice(index, 1, ...result.data.headingOrder);
+	}
+
+	// replace each iframe in the top-level heading order with its
+	// results.
+	// since nested iframe results can appear before their parent
+	// iframe, we will just loop over the nested results and
+	// piece-meal replace each iframe in the top-level heading order
+	// with their results until we no longer have results to replace
+	while (nestedResults.length) {
+		for (let i = 0; i < nestedResults.length; ) {
+			const nestedResult = nestedResults[i];
+			const index = getFrameIndex(nestedResult);
+
+			if (index !== -1) {
+				replaceFrameWithResults(index, nestedResult);
+
+				// remove the nested result from the list
+				nestedResults.splice(i, 1);
+			} else {
+				i++;
+			}
+		}
+	}
+
+	// replace the ancestry path with information about the result
+	results.forEach(result => {
+		const path = result.node.ancestry.join(joinStr);
+		const heading = headingOrder.find(heading => {
+			return heading.ancestry === path;
+		});
+		const index = headingOrder.indexOf(heading);
```

Comment:
Seems like this might work:

```suggestion
		const index = headingOrder.findIndex(heading => {
			return heading.ancestry === path;
		});
```

Reply:
`findIndex` is not supported in IE11 and I didn't want to bring in a polyfill to use it

Reply:
Fair enough!